src/main/java/org/openrewrite/java/testing/junit5/AssertThrowsOnLastStatement.java

```diff
@@ -106,37 +111,102 @@ public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration methodDecl
 
                     J.Block body = (J.Block) lambda.getBody();
                     List<Statement> lambdaStatements = body.getStatements();
-                    if (lambdaStatements.size() <= 1) {
-                        return methodStatement;
-                    }
 
                     // TODO Check to see if last line in lambda does not use a non-final variable
 
                     // move all the statements from the body into before the method invocation, except last one
-                    return ListUtils.map(lambdaStatements, (idx, lambdaStatement) -> {
+                    return ListUtils.flatMap(lambdaStatements, (idx, lambdaStatement) -> {
                         if (idx < lambdaStatements.size() - 1) {
-                            return lambdaStatement.withPrefix(methodStatement.getPrefix().withComments(Collections.emptyList()));
+                            return lambdaStatement.withPrefix(methodStatement.getPrefix().withComments(emptyList()));
                         }
 
+                        List<Statement> variableAssignments = new ArrayList<>();
+                        Space prefix = methodStatement.getPrefix().withComments(emptyList());
+                        final Statement newLambdaStatement = extractExpressionArguments(lambdaStatement, variableAssignments, prefix);
                         J.MethodInvocation newAssertThrows = methodInvocation.withArguments(
                                 ListUtils.map(arguments, (argIdx, argument) -> {
                                     if (argIdx == 1) {
                                         // Only retain the last statement in the lambda block
-                                        return lambda.withBody(body.withStatements(singletonList(lambdaStatement)));
+                                        return lambda.withBody(body.withStatements(singletonList(newLambdaStatement)));
                                     }
                                     return argument;
                                 })
                         );
 
                         if (assertThrowsWithVarDec == null) {
-                            return newAssertThrows;
+                            variableAssignments.add(newAssertThrows);
+                            return variableAssignments;
                         }
 
                         J.VariableDeclarations.NamedVariable newAssertThrowsVar = assertThrowsVar.withInitializer(newAssertThrows);
-                        return assertThrowsWithVarDec.withVariables(singletonList(newAssertThrowsVar));
+                        variableAssignments.add(assertThrowsWithVarDec.withVariables(singletonList(newAssertThrowsVar)));
+                        return variableAssignments;
                     });
                 })));
             }
+
+            private Statement extractExpressionArguments(Statement lambdaStatement, List<Statement> precedingVars, Space varPrefix) {
```

Comment:
Is there another way to fill the `precedingVars`? This method is now doing two things:
- getting the last call as statement
- filling the [optional] statements in the `precedingVars` variable

By doing two things at once, this method is harder to understand.